types/filters/parse.go

```diff
@@ -68,6 +69,26 @@ func ToParam(a Args) (string, error) {
 	return string(buf), nil
 }
 
+func ToParamWithVersion(version string, a Args) (string, error) {
+	// this way we don't URL encode {}, just empty space
+	if a.Len() == 0 {
+		return "", nil
+	}
+
+	// for daemons older than v1.10, filter must be of the form map[string][]string
+	buf := []byte{}
+	err := errors.New("")
+	if version != "" && compareTo(version, "1.22") == -1 {
```

Comment:
I think you mean "1.22 and above". It should be version `LessThan 1.22`, otherwise `1.23` will be wrong.